File: Enzo.Music.Kantaro.Test/UtToni_Semitoni.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Enzo.Music.Kantaro.Test;

internal class UtToni_Semitoni
{
    [Test]
    public void TestToni() {
        var t1 = Tono.Value;
        var t2 = Tono.Value;
        Assert.That(t2, Is.EqualTo(t1));
        Assert.That(t2, Is.SameAs(t1));
        var d = t1 + t2;
        Assert.IsTrue(d.Valore == 2);
    }

    [Test]
    public void TestSemitoni()
    {
        var v1 = Semitono.Value;
        var v2 = Semitono.Value;
        Assert.That(v2, Is.EqualTo(v1));
        Assert.That(v2, Is.SameAs(v1));
        Assert.That(Tono.Value, Is.EqualTo(v1 + v2));
        var d = v1 + v2;
        Assert.That(d.Valore, Is.EqualTo(1.0F));
        d = v1 * 2 + v2;
        Assert.That(d.Valore, Is.EqualTo(1.5F));
        d = 2 * v1 + v2;
        Assert.That(d.Valore, Is.EqualTo(1.5F));
    }
}

File: MusicTestProject/utToni_Semitoni.vb
Imports System.Text
Imports Microsoft.VisualStudio.TestTools.UnitTesting

<TestClass()>
Public Class utToni_Semitoni

    <TestMethod()>
    Public Sub TestToni()
        Dim t1 = Tono.Value
        Dim t2 = Tono.Value
        Assert.AreEqual(t1, t2)
        Assert.AreSame(t1, t2)
        Dim d = t1 + t2
        Assert.IsTrue(d.Valore = 2)
    End Sub

    <TestMethod()>
    Public Sub TestSemitoni()
        Dim v1 = Semitono.Value
        Dim v2 = Semitono.Value
        Assert.AreEqual(v1, v2)
        Assert.AreSame(v1, v2)
        Assert.AreEqual(v1 + v2, Tono.Value)
        Dim d = v1 + v2
        Assert.AreEqual(1.0!, d.Valore)
        d = v1 * 2 + v2
        Assert.AreEqual(1.5!, d.Valore)
        d = 2 * v1 + v2
        Assert.AreEqual(1.5!, d.Valore)
    End Sub

End Class
